onchain/permissionless-arbitration/src/Match.sol

```diff
@@ -157,17 +158,25 @@ library Match {
     }
 
     function getDivergence(
-        State storage state
+        State storage state,
+        uint256 startCycle
     )
         internal
         view
-        returns (Machine.Hash finalStateOne, Machine.Hash finalStateTwo)
+        returns (
+            Machine.Hash agreeHash,
+            uint256 agreeCycle,
+            Machine.Hash finalStateOne,
+            Machine.Hash finalStateTwo
+        )
     {
         assert(state.currentHeight == 0);
+        agreeHash = Machine.Hash.wrap(Tree.Node.unwrap(state.otherParent));
+        agreeCycle = state.toCycle(startCycle);
```

Comment:
Good question, I think it's right (I'm not sure), but it's subtle. Tests should give us more certainty.

First, remember that the tree doesn't include the initial hash of the tournament. So we have this initial hash which everyone agrees with, corresponding to cycle `0`, and a tree where players may disagree on the very first leaf.

Imagine that's the case. Then we'll always go down left on the tree, never incrementing `runningLeafPosition`. It will be `0`, representing that the players disagree on leaf `0`. Then, `toCycle` will return `0`, which refers to the initial hash not contained in the tree.

It seems like there indices conversions which are making things harder to understand. Did this explanation make sense? I'm not completely sure haha.

Reply:
You're right, I don't know that the initial hash is not included in the tree. Which makes sense too, it should never be part of the dispute in the first place.

Reply:
Completely agree! After we have something running with step, reset, etc., we'll have to dedicate some effort to improve code quality and documentation.

I think it could be a good exercise, after we've added minimal documentation, to have an internal audit and use the feedback to understand where we need to improve.